### DISCUSSION TITLE
Do prisma _count queries count as 1 row read per COUNT?

### DISCUSSION THREAD

**Discussion by jacoblatona** [planetscale/discussion#228] [Q&A] [answered]
Hello, 

I have a question regarding how row reads are calculated when having a _count parameter in your prisma query. 

So for example, let's say this blog post has 20 comments. 

`const blog = await prisma.blog.findFirst({
     where: { id: id }
     include: {
       _count: {
         select: { comments: true }
       }
    }.  
})`

Will this query count as 20 row reads since there were 20 comments that were counted?

Thank you very much for your time.

**Comment by janpio** [comment, marked as answer]
What is the SQL that Prisma generates for this query under the hood? https://pris.ly/d/logging should allow you to get that - I assume PlanetScale will need that to be able to evaluate this question.

**Reply by jacoblatona** [reply]
Hi, janpio! 

I just looked at the Planetscale FAQ more in-depth and it seems they answer my question. 

Raw COUNT queries do not in-fact count as reads. My apologies I should have read the docs more in depth before posting this question.

**Reply by orware** [reply]
Thank you for the update @jacoblatona on your question here!

For others visiting this page, the reference link in our Docs is the following one:
https://docs.planetscale.com/concepts/billing#caveats

**Comment by zigazajc007** [comment]
Hello,

I have just read the caveats section that you have shared.

> The SELECT count(*) query is a special case. The database engine optimizes this query and doesn't read row data itself to return a count, so this doesn't actually increment rows read. In other words, if you run the query SELECT count(*) on a table with 10m rows, your rows read will be 0.

My SELECT COUNT(*) query does take quite a lot time. It seems that database engine didn't optimized this query. Will this still be counted as 0 rows read?
![image](https://user-images.githubusercontent.com/44822563/195345045-882ce234-3235-4211-bdda-be9d745f87cb.png)

Table passwords currently includes 14295 rows.
